### PR TITLE
CASSANDRA-16625 (4.0): Add a CircleCI job to run some tests repeatedly

### PR DESCRIPTION
**Pull request by adelapena**

### CODE REVIEW

File: .circleci/config-2_1.yml [hunk 1]
```diff
@@ -1102,3 +1207,174 @@ commands:
...
+                fi
+
+                # Prepare the -Dtest.methods argument, which is optional
+                if [ "$REPEATED_UTEST_METHODS" == "<nil>" ]; then
+                  methods=""
+                else
+                  methods="-Dtest.methods=$REPEATED_UTEST_METHODS"
+                fi
+
+                # Run the test target as many times as requested, even if it fails, collecting the exit code.
```
Comment (bereng) [inline .circleci/config-2_1.yml]: Maybe you can adjust this comment upon commit now that we have `REPEATED_UTEST_STOP_ON_FAILURE`

File: .circleci/config-2_1.yml [hunk 2]
```diff
@@ -1102,3 +1207,174 @@ commands:
...
           path: ~/cassandra-dtest/logs
           destination: dtest_<<parameters.file_tag>>_logs
+
+  run_repeated_utest:
+    steps:
+      - run:
+          name: Run repeated utest
+          no_output_timeout: 15m
+          command: |
+            if [ "$REPEATED_UTEST_CLASS" == "<nil>" ]; then
```
Comment (bereng) [inline .circleci/config-2_1.yml:1217]: Where does this `<nil>`come from?

Comment (adelapena) [reply .circleci/config-2_1.yml:1217]: That's what CircleCI puts on empty env vars. I think that the `nil` comes from Go.

Comment (bereng) [reply .circleci/config-2_1.yml:1217]: sgtm thx.

**Pull request closed without merging**